tests/brevitas/graph/test_gpxq.py

```diff
@@ -0,0 +1,143 @@
+        accumulator_bit_width: int = 32,
+        a2q_layer_filter_fnc=lambda x: True):
+    model.eval()
+    dtype = next(model.parameters()).dtype
+    device = next(model.parameters()).device
+    with torch.no_grad():
+        # use A2GPFQ if accumulator is less than 32 is specified
+        with gpfq_mode(
+                model,
+                use_quant_activations=use_quant_activations,
+                act_order=act_order,
+                use_gpfa2q=accumulator_bit_width < 32,
+                accumulator_bit_width=accumulator_bit_width,
+                a2q_layer_filter_fnc=a2q_layer_filter_fnc,
+        ) as gpfq:
+            gpfq_model = gpfq.model
+            for _ in range(gpfq.num_layers):
+                for _, (images, _) in enumerate(calib_loader):
+                    images = images.to(device)
+                    images = images.to(dtype)
+                    gpfq_model(images)
+                gpfq.update()
+
+
+def apply_gptq(
+        calib_loader: DataLoader, model: nn.Module, act_order: bool, use_quant_activations: bool):
+    model.eval()
+    dtype = next(model.parameters()).dtype
+    device = next(model.parameters()).device
+    with torch.no_grad():
+        with gptq_mode(
+                model,
+                use_quant_activations=use_quant_activations,
+                act_order=act_order,
+        ) as gptq:
+            gptq_model = gptq.model
+            for _ in range(gptq.num_layers):
+                for _, (images, _) in enumerate(calib_loader):
+                    images = images.to(device)
+                    images = images.to(dtype)
+                    gptq_model(images)
+                gptq.update()
+
+
+def custom_layer_filter_fnc(layer: nn.Module) -> bool:
+    if isinstance(layer, nn.Conv2d) and layer.in_channels == 3:
+        return False
+    elif isinstance(layer, nn.ConvTranspose2d) and layer.in_channels == 3:
+        return False
+    return True
+
+
+def identity_layer_filter_func(layer: nn.Module) -> bool:
+    return True
+
+
+filter_func_dict = {"identity": identity_layer_filter_func, "ignore_input": custom_layer_filter_fnc}
+
+apply_gpxq_func_map = {"gpfq": apply_gpfq, "gptq": apply_gptq}
+
+
+@pytest.mark.parametrize("act_order", [True, False])
+@pytest.mark.parametrize("use_quant_activations", [True, False])
+@pytest.mark.parametrize("acc_bit_width", [32, 24, 16, 12])
+@pytest.mark.parametrize("filter_func_str", filter_func_dict.keys())
+@pytest.mark.parametrize("apply_gpxq_tuple", apply_gpxq_func_map.items())
+def test_toymodels(
+        toy_quant_model,
+        act_order,
+        use_quant_activations,
+        acc_bit_width,
+        filter_func_str,
+        apply_gpxq_tuple,
+        request):
+
+    test_id = request.node.callspec.id
+
+    torch.manual_seed(SEED)
+
+    name, apply_gpxq = apply_gpxq_tuple
+
+    if (name == 'gptq' and acc_bit_width < 32):
+        pytest.skip("GPTQ does not support accumulator-aware quantization.")
```

Comment:
I would skip tests only when the testing configuration does not make sense.

For this case, I would raise an error in the code and catch it here with `pytest.raises`.

Similar for the one below.